argus/shell.py

```diff
@@ -0,0 +1,261 @@
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-i", "--image_id", action="append",
+                       help="Glance image-id to use for this run.")
+    group.add_argument("-n", "--image_name", action="append",
+                       help="Glance image-name to use for this run."
+                            "This is dangerous, be sure the name matches"
+                            " just one image.")
+    parser.add_argument("-f", "--flavor",
+                        help="The flavor id.")
+    parser.add_argument("-t", "--tests",
+                        help="The tests you want to run.")
+    parser.add_argument("-r", "--resources",
+                        help="URL to argus resources.")
+    parser.add_argument("-p", "--parallel", type=int,
+                        help="many processes to use in parallel.")
+    parser.add_argument("-l", "--local",
+                        help="Local git repository with all the argus files.")
+    parser.add_argument("-s", "--separate", action="store_true",
+                        help="Make sepatate log files for each scenario.")
+    parser.add_argument("-g", "--git_command",
+                        help="The git command that will test the "
+                             "Cloudbase-Init patch.")
+    parser.add_argument("-m", "--msi", help="The msi to use.")
+    return parser
+
+
+def _prepare_enviroment(local, directory):
+    """Prepare the temp enviroment."""
+    os.mkdir(os.path.join(directory, "ci"))
+
+    if local:
+        local = os.path.abspath(local)
+        os.link(os.path.join(local, ".testr.conf"),
+                os.path.join(directory, ".testr.conf"))
+
+        os.link(os.path.join(local, "ci", "tests.py"),
+                os.path.join(directory, "ci", "tests.py"))
+    else:
+        # Download the necessary items
+        download_argus_resource(".testr.conf", os.path.join(
+            directory, ".testr.conf"))
+
+        download_argus_resource("ci/tests.py", os.path.join(
+            directory, "ci", "tests.py"))
+
+    # Create a new repository
+    proc = subprocess.Popen(["testr", "init"], cwd=directory, close_fds=True)
+    proc.wait()
+
+
+def _prepare_config(separate, resources, flavor,
+                    git_command, msi,
+                    directory, image_id):
+    """Prepare the argus config file."""
+
+    conf = six.moves.configparser.SafeConfigParser()
+    conf.add_section("argus")
+    conf.add_section("openstack")
+
+    conf.set("argus", "output_directory", os.path.join(directory, "output"))
+    conf.set("argus", "argus_log_file", os.path.join(directory, "argus.log"))
+
+    if resources:
+        conf.set("argus", "resources", str(resources))
+
+    if separate:
+        conf.set("argus", "individual_logs", str(separate))
+
+    if flavor:
+        conf.set("openstack", "flavor_id", str(flavor))
+
+    conf.set("openstack", "image_ref", str(image_id))
+    conf.set("argus", "git_command", str(git_command))
```

Comment:
Please change this code block to: 
`if git_command:
    conf.set("argus", "git_command", str(git_command))
`
And the same thing for the `msi` which we will need to change